tensorflow/core/util/gpu_kernel_helper.h

```diff
@@ -18,61 +18,70 @@ limitations under the License.
 
 #if GOOGLE_CUDA || TENSORFLOW_USE_ROCM
 
-#if GOOGLE_CUDA
-#include "cuda/include/cuda_fp16.h"
-#endif
+#include "tensorflow/core/util/gpu_cuda_alias.h"
 #include "tensorflow/core/util/gpu_device_functions.h"
 #include "tensorflow/core/util/gpu_launch_config.h"
 
+#if GPU_VERSION >= 7050
+#include "cuda/include/cuda_fp16.h"
+#define TF_HAS_GPU_FP16
+#endif
+
 #if GOOGLE_CUDA
 #define TF_RED_WARPSIZE 32
 #elif TENSORFLOW_USE_ROCM
 #define TF_RED_WARPSIZE 64
 #endif
 
 #if GOOGLE_CUDA
-#define GPU_LAUNCH_KERNEL(kernel, block_count, threads_per_block, shared_mem, \
-                          stream, ...)                                        \
-  TF_CHECK_OK(CudaLaunchKernel(kernel, block_count, threads_per_block,        \
-                               shared_mem, stream, __VA_ARGS__));
+#define GPU_LAUNCH_KERNEL(kernel, block_count, threads_per_block, \
+                          shared_mem, stream, ...) \
+  kernel<<<block_count, threads_per_block, shared_mem, stream>>>(__VA_ARGS__);
```

Comment:
@jerryyin per discussion we'll make `GPU_LAUNCH_KERNEL` macro obsolete and introduce `GpuLaunchKernel`, so maybe we can ignore adding more changes to this macro in this PR.